### PR TITLE
doc: better security rules documentation (#1305)

### PR DESCRIPTION
**Pull request by RiccardoManzan**
Here's the docs addition described in #1305. I tried executing `./gradlew publishGuide` but i get a "build failed". I guess this is due to a local misconfiguration.

### CODE REVIEW

File: src/main/docs/guide/securityRule.adoc [hunk 1]
```diff
@@ -1,28 +1,44 @@
 The decision to allow access to a particular endpoint to anonymous or authenticated users is determined by a collection of
-Security Rules. Micronaut ships with several built-in security rules. If they don't fulfil your needs,
+Security Rules which are executed from the api:security.filter.SecurityFilter[]. Micronaut ships with several built-in security rules. If they don't fulfil your needs,
```
Comment (sdelamo) [inline src/main/docs/guide/securityRule.adoc:2]: The package is `io.micronaut.security.filters`. You missed the trailing `s`. 

```suggestion
Security Rules which are executed from the api:security.filters.SecurityFilter[]. Micronaut ships with several built-in security rules. If they don't fulfil your needs,
```

File: src/main/docs/guide/securityRule.adoc [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
-|Access to the resource should be rejected and no further rules will be considered.
+|Access to the resource should be rejected and *no further rules will be considered*.
 
 |api:security.rules.SecurityRuleResult#UNKNOWN[UNKNOWN]
 |The rule doesn't apply to the request resource, or it cannot be determined either way. This result will cause other security rules to be considered.
 |===
 
 IMPORTANT: If all security rules return `UNKNOWN`, the request will be rejected!
 
+IMPORTANT: This implementation of api:security.filter.SecurityFilter[] so the security rules are exclusive: The first that manages the request (which returns alternatively `ALLOWED` or `REJECTED`) will be *the only one considered*.
```
Comment (sdelamo) [inline src/main/docs/guide/securityRule.adoc:22]:
```suggestion
IMPORTANT: This implementation of api:security.filters.SecurityFilter[] so the security rules are exclusive: The first that manages the request (which returns alternatively `ALLOWED` or `REJECTED`) will be *the only one considered*.
```

Comment (sdelamo) [inline src/main/docs/guide/securityRule.adoc:22]:
```suggestion
IMPORTANT: api:security.filter.SecurityFilter[] evaluates security rules in order. **The remaining rules are not evaluated once a rule returns `ALLOWED` or `REJECTED`**.
```

File: src/main/docs/guide/securityRule/interceptUrlMap.adoc [hunk 3]
```diff
@@ -32,3 +32,5 @@ micronaut:
 
 - accessing `/v1/myResource/**` with a GET request does not require authentication
 - accessing `/v1/myResource/**` with a request that isn't GET requires authentication
+
+IMPORTANT: When a ann:security.annotation.Secured[] annotation is provided with a set of roles, then a user will have to access to that resource if they have *at least one* of thoose roles, not all.
```
Comment (sdelamo) [inline src/main/docs/guide/securityRule/interceptUrlMap.adoc:36]:
```suggestion
IMPORTANT: When the ann:security.annotation.Secured[] annotation has a set of roles, the <<secured, SecuredAnnotationRule>> grants access to a user if they have **any** of the roles.
```

File: src/main/docs/guide/securityRule/secured.adoc [hunk 4]
```diff
@@ -26,3 +26,4 @@ include::{testssecurityjava}/security/securityRule/permitall/ExampleController.j
 
 NOTE: The use of JSR 250 annotations requires `io.micronaut.security:micronaut-security-annotations` to be in the annotation processor classpath (`annotationProcessor`, `kapt`, `compileOnly`) respectively for Java, Kotlin, Groovy.
 
+IMPORTANT: When a ann:security.annotation.Secured[] annotation is provided with a set of roles, then a user will have to access to that resource if they have *at least one* of thoose roles, not all.
```
Comment (sdelamo) [inline src/main/docs/guide/securityRule/secured.adoc:29]:
```suggestion
IMPORTANT: When the ann:security.annotation.Secured[] annotation has a set of roles, the <<secured, SecuredAnnotationRule>> grants access to a user if they have **any** of the roles.
```

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Comment by sdelamo** [comment]
Thanks for the PR @RiccardoManzan I have provided several suggestions which you can commit directly from the Github UI. 

commit suggestions

File: src/main/docs/guide/securityRule.adoc (hunk 2 above)
Comment (RiccardoManzan) [reply src/main/docs/guide/securityRule.adoc:22]: What I wrote here actually didn't have any sense. I guess I've got tricked by copy and paste 😅

**Review by sdelamo — APPROVED** [review]

**Pull request merged**